tinyxml2 port ships malformed pkg-config .pc on x64-linux

**Describe the bug**
tinyxml2 port ships malformed pkg-config .pc, in particular the file `installed/x64-linux/debug/lib/pkgconfig/tinyxml2.pc` is malformed, as it references to the include directory `/home/straversaro/src/vcpkg/packages/tinyxml2_x64-linux/debug/include` that do not actually contain any include.
~~~
straversaro@iiticublap103:~/src/vcpkg$ cat ./installed/x64-linux/debug/lib/pkgconfig/tinyxml2.pc 
prefix=/home/straversaro/src/vcpkg/packages/tinyxml2_x64-linux/debug
exec_prefix=${prefix}
libdir=${exec_prefix}/lib
includedir=${prefix}/include

Name: TinyXML2
Description: simple, small, C++ XML parser
Version: 7.0.1
Libs: -L${libdir} -ltinyxml2
Cflags: -I${includedir}
~~~

**Environment**
- OS: Ubuntu 18.04 
- Compiler: revision

**To Reproduce**
Steps to reproduce the behavior:
1. ./vcpkg install tinyxml2
2. See malformed pkg-config file

**Expected behavior**
The pkg-config .pc file should contain valid directories, or not be shipped otherwise.

**Failure logs** 
While configuring the ignition-msgs5 project (see https://bitbucket.org/ignitionrobotics/ign-msgs/src/ignition-msgs5_5.1.0/):
~~~
straversaro@iiticublap103:~/src/vcpkg$ cat  /home/straversaro/src/vcpkg/buildtrees/ignition-msgs5/config-x64-linux-dbg-err.log
CMake Error in src/CMakeLists.txt:
  Imported target "TINYXML2::TINYXML2" includes non-existent path

    "/home/straversaro/src/vcpkg/packages/tinyxml2_x64-linux/debug/include"

  in its INTERFACE_INCLUDE_DIRECTORIES.  Possible reasons include:

  * The path was deleted, renamed, or moved to another location.

  * An install or uninstall procedure did not complete successfully.

  * The installation package was faulty and references files it does not
  provide.
~~~

**Additional context**
This problem may be fixed by using the function introduced in https://github.com/microsoft/vcpkg/pull/9861 in the `tinyxml2` port, but I am not sure about this.

I tried to fix this by adding `vcpkg_fixup_pkgconfig`, but in that case an additional error is reported:
~~~
Building package tinyxml2[core]:x64-linux...
-- Using cached /home/straversaro/vcpkg/downloads/leethomason-tinyxml2-9c740e8d2341bd46dbe8e87053cdb4d931971967.tar.gz
-- Using source at /home/straversaro/vcpkg/buildtrees/tinyxml2/src/d931971967-36d1de7ae2
-- Configuring x64-linux-dbg
-- Configuring x64-linux-rel
-- Building x64-linux-dbg
-- Building x64-linux-rel
-- Fixing pkgconfig
-- Fixing pkgconfig - release
-- Checking file: /home/straversaro/vcpkg/packages/tinyxml2_x64-linux/lib/pkgconfig/tinyxml2.pc
-- Fixing pkgconfig - debug
-- Checking file: /home/straversaro/vcpkg/packages/tinyxml2_x64-linux/debug/lib/pkgconfig/tinyxml2.pc
CMake Error at scripts/cmake/vcpkg_fixup_pkgconfig.cmake:122 (message):
  Found
  /home/straversaro/vcpkg/packages/tinyxml2_x64-linux/debug/lib/libtinyxml2d.a
  with additional debug suffix! Please correct the *.pc file!
Call Stack (most recent call first):
  scripts/cmake/vcpkg_fixup_pkgconfig.cmake:261 (vcpkg_fixup_pkgconfig_check_libraries)
  ports/tinyxml2/portfile.cmake:20 (vcpkg_fixup_pkgconfig)
  scripts/ports.cmake:90 (include)


Error: Building package tinyxml2:x64-linux failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `.\vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: tinyxml2:x64-linux
  Vcpkg version: 2020.02.04-unknownhash
~~~

--- a/ports/tinyxml2/CONTROL
+++ b/ports/tinyxml2/CONTROL
@@ -1,4 +1,4 @@
 Source: tinyxml2
-Version: 8.0.0
+Version: 8.0.0-1
 Homepage: https://github.com/leethomason/tinyxml2
 Description: A simple, small, efficient, C++ XML parser

--- a/ports/tinyxml2/fix-debug-postfix.patch
+++ b/ports/tinyxml2/fix-debug-postfix.patch
@@ -1,0 +1,43 @@
+From 1e384d50313a08dd36a43df1925588741dc33301 Mon Sep 17 00:00:00 2001
+From: =?UTF-8?q?Wolfgang=20St=C3=B6ggl?= <c72578@yahoo.de>
+Date: Wed, 17 Jun 2020 06:17:06 +0200
+Subject: [PATCH] Fix debug postfix in pkgconfig file
+
+CMakeLists.txt uses set(CMAKE_DEBUG_POSTFIX "d") to distinguish
+between debug and release lib. Use this postfix also in the
+generated tinyxml2.pc file in case of a CMake Debug build.
+This results in the following contents of tinyxml2.pc
+- Release:
+  Libs: -L${libdir} -ltinyxml2
+- Debug:
+  Libs: -L${libdir} -ltinyxml2d
+---
+ CMakeLists.txt | 3 +++
+ tinyxml2.pc.in | 2 +-
+ 2 files changed, 4 insertions(+), 1 deletion(-)
+
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index 9f9ab07..7085276 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -106,6 +106,9 @@ endif()
+ 
+ install(FILES tinyxml2.h DESTINATION ${CMAKE_INSTALL_INCLUDEDIR} COMPONENT tinyxml2_headers)
+ 
++if(CMAKE_BUILD_TYPE MATCHES Debug)
++  set(LIB_POSTFIX ${CMAKE_DEBUG_POSTFIX})
++endif()
+ configure_file(tinyxml2.pc.in tinyxml2.pc @ONLY)
+ install(FILES ${CMAKE_CURRENT_BINARY_DIR}/tinyxml2.pc DESTINATION ${CMAKE_INSTALL_LIBDIR}/pkgconfig COMPONENT tinyxml2_config)
+ 
+diff --git a/tinyxml2.pc.in b/tinyxml2.pc.in
+index b040b0e..b100f60 100644
+--- a/tinyxml2.pc.in
++++ b/tinyxml2.pc.in
+@@ -6,5 +6,5 @@ includedir=${prefix}/@CMAKE_INSTALL_INCLUDEDIR@
+ Name: TinyXML2
+ Description: simple, small, C++ XML parser
+ Version: @GENERIC_LIB_VERSION@
+-Libs: -L${libdir} -ltinyxml2
++Libs: -L${libdir} -ltinyxml2@LIB_POSTFIX@
+ Cflags: -I${includedir}

--- a/ports/tinyxml2/portfile.cmake
+++ b/ports/tinyxml2/portfile.cmake
@@ -6,6 +6,7 @@ vcpkg_from_github(
     HEAD_REF master
     PATCHES
         fix-symbol-exports.patch
+        fix-debug-postfix.patch
 )
 
 vcpkg_configure_cmake(
@@ -16,6 +17,7 @@ vcpkg_configure_cmake(
 vcpkg_install_cmake()
 
 vcpkg_fixup_cmake_targets(CONFIG_PATH lib/cmake/tinyxml2)
+vcpkg_fixup_pkgconfig()
 
 vcpkg_copy_pdbs()
 

